lib/resty/upstream/healthcheck.lua

```diff
@@ -602,6 +606,16 @@ function _M.spawn_checker(opts)
         concurrency = concur,
     }
 
+    -- as far as we do not support https override ports if set
+    if http_port then
+        for i, _ in ipairs(ctx.primary_peers) do
+            ctx.primary_peers[i].port = http_port
```

Comment:
Hmm, I'm not sure about this. The user may explicitly specify a port number in the `server` directive in `upstream {}` blocks. For example,

```nginx
    upstream {
        server 10.2.136.5:8080;
        server 10.2.136.6:8080;
    }
```

See http://nginx.org/en/docs/http/ngx_http_upstream_module.html#server for more details.